packages/client/src/BrubeckNode.ts

```diff
@@ -62,6 +64,14 @@ export default class BrubeckNode implements Context {
         const { options } = this
         let { id } = options
 
+        if ((options.trackers as TrackerRegistrySmartContract).contractAddress) {
```

Comment:
Could extract this to a method called `getNormalizedNetworkOptions()` which would not modify the this.options but return a modified clone? That way we could keep `this.options` as an immutable variable.